src/include/duckdb/function/table/arrow/arrow_duck_schema.hpp

```diff
@@ -0,0 +1,103 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/function/table/arrow_duck_schema.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <utility>
+
+#include "duckdb/common/types.hpp"
+#include "duckdb/common/unordered_map.hpp"
+
+namespace duckdb {
+//===--------------------------------------------------------------------===//
+// Arrow Variable Size Types
+//===--------------------------------------------------------------------===//
+enum class ArrowVariableSizeType : uint8_t { FIXED_SIZE = 0, NORMAL = 1, SUPER_SIZE = 2 };
+
+//===--------------------------------------------------------------------===//
+// Arrow Time/Date Types
+//===--------------------------------------------------------------------===//
+enum class ArrowDateTimeType : uint8_t {
+	MILLISECONDS = 0,
+	MICROSECONDS = 1,
+	NANOSECONDS = 2,
+	SECONDS = 3,
+	DAYS = 4,
+	MONTHS = 5,
+	MONTH_DAY_NANO = 6
+};
+
+class ArrowType {
+public:
+	// Move constructor
+	ArrowType(ArrowType &&other) = default;
```

Comment:
I think it's better to have `unique_ptr`than move constructors. Since unique_ptrs get properly invalidated.